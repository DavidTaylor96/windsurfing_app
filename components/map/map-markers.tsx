import { Fontisto } from '@expo/vector-icons';
import React, { FC } from 'react';
import { Marker } from 'react-native-maps';
import { useMarkerPressed } from '../../hooks/marker-pressed';
import { useMapNavigation } from '../../hooks/use-map-navigation';
import { markers } from './map-data';

interface IMapMarkersProps {}

export const MapMarkers: FC<IMapMarkersProps> = ({}) => {

  const { pressed, setPressed } = useMarkerPressed();

  const {animateToCoordinat } = useMapNavigation();
  return (
    <>
      {markers.map((marker) => (
        <Marker
          onPress={() => {
            animateToCoordinat(marker.latitude, marker.longitude);
            setPressed(!pressed);
          }}
          key={marker.id}
          identifier={marker.id}
          coordinate={{
            latitude: marker.latitude,
            longitude: marker.longitude,
          }}
          title={marker.title}
        >
          <Fontisto name="map-marker-alt" size={29} color="#bf0f2b" />
        </Marker>
      ))}
    </>
  );
};
