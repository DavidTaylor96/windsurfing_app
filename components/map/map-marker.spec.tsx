import { Fontisto } from '@expo/vector-icons';
import * as React from 'react';
import renderer from 'react-test-renderer';
import { MapMarkers } from './map-markers';


it(`Map markers`, () => {


  const mapMarker = renderer.create(
    <MapMarkers>
      <Fontisto name="map-marker-alt" size={29} color="#bf0f2b" />
    </MapMarkers>
  );

  expect(mapMarker.root.children.length).toBe(3);

  expect(mapMarker).toMatchSnapshot();
});
