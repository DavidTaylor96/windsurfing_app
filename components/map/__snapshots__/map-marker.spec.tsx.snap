// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map markers 1`] = `
Array [
  <AIRMapMarker
    coordinate={
      Object {
        "latitude": 55.953251,
        "longitude": -3.188267,
      }
    }
    identifier="123"
    onPress={[Function]}
    stopPropagation={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
    title="one"
  >
    <Text />
  </AIRMapMarker>,
  <AIRMapMarker
    coordinate={
      Object {
        "latitude": 55.853251,
        "longitude": -2.488267,
      }
    }
    identifier="124"
    onPress={[Function]}
    stopPropagation={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
    title="two"
  >
    <Text />
  </AIRMapMarker>,
  <AIRMapMarker
    coordinate={
      Object {
        "latitude": 55.753251,
        "longitude": -2.588267,
      }
    }
    identifier="125"
    onPress={[Function]}
    stopPropagation={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
    title="Three"
  >
    <Text />
  </AIRMapMarker>,
]
`;
